From: Matthew Wilcox <matthew.r.wilcox@intel.com>
Date: Fri, 7 Oct 2011 13:10:13 -0400
Subject: NVMe: Don't probe namespace 0
Git-commit: 2b2c1896871838cdf549442e8ad0264be5fa74e3
References: FATE#313627
Patch-Mainline: v3.7-rc1

ECN 001 documented that namespace 0 is not valid.  Sending an Identify
with CNS of 0 and Namespace of 0 is an undefined command.

Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/block/nvme.c b/drivers/block/nvme.c
index 3afdc75..660aa5d 100644
--- a/drivers/block/nvme.c
+++ b/drivers/block/nvme.c
@@ -1447,7 +1447,7 @@ static int __devinit nvme_dev_add(struct nvme_dev *dev)
 	memcpy(dev->firmware_rev, ctrl->fr, sizeof(ctrl->fr));
 
 	id_ns = mem;
-	for (i = 0; i <= nn; i++) {
+	for (i = 1; i <= nn; i++) {
 		res = nvme_identify(dev, i, 0, dma_addr);
 		if (res)
 			continue;
-- 
1.7.4.2

